### ISSUE TITLE
Spectator view should defer loading large assets until needed.

### ISSUE THREAD

**Issue by chrisfromwork** [microsoft/MixedReality-SpectatorView#197]
Spectator view loads asset caches initially, so if a lot of meshes/large memory assets are included in asset caches, you can hit a point where the app crashes based on a lack of memory. We should assess whether loading these assets can be deferred until they are needed.

**Comment by emartinDesignInteractive** [comment]
Any update on this? It is preventing me from progressing using Spectator View

**Comment by chrisfromwork** [comment]
We are looking into this now.

**Comment by RealFrogee** [comment]
After I run "Update All Asset Caches" in my project, the project build size has increased by 1G. And when I open it on Hololens,it crashed.I have tested several situations.Without "Update All Asset Caches" the build can run normally on Hololens.
Is the crash caused by the large assets problem?Is there a solution now?

**Comment by chrisfromwork** [comment]
@RealFrogee the issue is based on how asset caches currently work. Basically all assets in your unity project get declared in an asset cache. Then when the hololens starts up it tries to load all asset caches. This is okay for small unity projects/targetted demos, but is really bad for real world applications whose projects contain a lot of content. The review we have out in #295 may be able to unblock you. It still has the same amount of asset references in a resource folder, which i would imagine will still increase your project build size. But compared to attempting to load all assets on start, a list of asset names is loaded. Then when spectator view observes assets declared in your scene, this asset name list is used to load individual asset references. There will still be some caveats, if you have any assets with the same name and type, all of them will be loaded at the same time even if you are only using one of them.
